packages/contracts-noir/test/Semaphore.ts

```diff
@@ -0,0 +1,672 @@
+        })
+
+        it("Should not verify a proof if the Merkle tree root is not part of the group", async () => {
+            const { semaphoreContract, groupId, proof } = await loadFixture(deployVerifyProofFixture)
+
+            const transaction = semaphoreContract.verifyProof(groupId, { ...proof, merkleTreeRoot: 1 })
+
+            await expect(transaction).to.be.revertedWithCustomError(
+                semaphoreContract,
+                "Semaphore__MerkleTreeRootIsNotPartOfTheGroup"
+            )
+        })
+
+        it("Should verify a proof for an onchain group", async () => {
+            const { semaphoreContract, groupId, proof } = await loadFixture(deployVerifyProofFixture)
+            // use static call so we can retrieve the value of a non-view function
+            const validProof = await semaphoreContract.verifyProof.staticCall(groupId, proof)
+            expect(validProof).to.equal(true)
+        })
+
+        it("Should not verify a proof if the Merkle tree root is expired", async () => {
+            const { semaphoreContract, accountAddresses, members } = await loadFixture(deployVerifyProofFixture)
+
+            // create new group with 0s Merkle tree root expiration
+            const groupId = 1
+            await semaphoreContract["createGroup(address,uint256)"](accountAddresses[0], 0)
+            await semaphoreContract.addMember(groupId, members[0])
+            await semaphoreContract.addMember(groupId, members[1])
+            await semaphoreContract.addMember(groupId, members[2])
+
+            const message = 2
+            const merkleTreeDepth = 2
+            const identity = new Identity("0")
+            const group = new Group()
+
+            group.addMembers([members[0], members[1]])
+
+            const proof = await generateNoirProof(
+                identity,
+                group,
+                message,
+                group.root,
+                merkleTreeDepth,
+                undefined,
+                true
+            )
+
+            const transaction = semaphoreContract.verifyProof(groupId, proof)
+
+            await expect(transaction).to.be.revertedWithCustomError(
+                semaphoreContract,
+                "Semaphore__MerkleTreeRootIsExpired"
+            )
+        })
+
+        it("Should not verify a proof if the Merkle depth is not supported", async () => {
+            const { semaphoreContract, groupId, members } = await loadFixture(deployVerifyProofFixture)
+
+            const message = 2
+            const merkleTreeDepth = 12
+            const identity = new Identity("0")
+            const group = new Group()
+
+            group.addMembers(members)
+
+            const scope = "random-scope"
+            const proof = await generateNoirProof(identity, group, message, scope, merkleTreeDepth, undefined, true)
+            proof.proofBytes = proof.proofBytes.slice(4)
```

Comment:
please add a comment as to why this is needed (also, this is a good question for Noir team)